src/cpp/src/llm_pipeline.cpp

```diff
@@ -89,6 +90,8 @@ class StatefulLLMPipeline final : public LLMPipelineImplBase {
         // If eos_token_id was not provided, take value
         if (m_generation_config.eos_token_id == -1)
             m_generation_config.set_eos_token_id(m_tokenizer.get_eos_token_id());
+
+        sampler = Sampler(m_tokenizer);
```

Comment:
let's call `m_sampler.set_state(m_generation_config.rng_state)` similarly to 
https://github.com/openvinotoolkit/openvino.genai/blob/d465c5bc43c706459ebba4b02925b57cb7bbb56f/src/cpp/src/continuous_batching_impl.cpp#L77

In general, it's not very clear whether we need to reset_state each `generate` call to `GenerationConfig::rng_state` as user can write:
```c++
pipe.generate("123", seed=1);
pipe.generate("456", seed=12);
```
@sbalandi @Wovchena how do you think?
What is HF behavior?

Reply:
hf manages this via call of function set_seed(), it can be called multiple times between generate(), so we can also overwrite seed, if the value have changed

Reply:
I'd guess  resetting seed on every `geneare()` would result in the same generated results just like in the original problem. But preserving the requested seed in constructor seems to be fine

Reply:
maybe we can track if initial seed for `Sampler` (set in Pipeline ctor via `Sampler::set_seed`) is not the same as in `generate()`, it means user wants to change it?

Reply:
Makes sense to me

Reply:
moved behavior to: if seed in generation_config passed to generate is not the same as sampler -> change it